proposals/NNNN-package-manager-swift-language-compatibility-version.md

```diff
@@ -0,0 +1,143 @@
+# Package Manager Swift Language Compatibility Version
+
+* Proposal: [SE-NNNN](NNNN-package-manager-swift-language-compatibility-version.md)
+* Author: [Daniel Dunbar](https://github.com/ddunbar), [Rick Ballard](http://github.com/rballard)
+* Review manager: Anders Bertelrud
+* Status: Scheduled for review (February 7...February 13, 2017)
+
+## Introduction
+
+This proposal adds support for the Swift compiler's new "language compatibility
+version" feature to the package manager.
+
+## Motivation
+
+The Swift compiler now supports a "language compatibility version" flag which
+specifies the Swift major language version that the compiler should try to
+accept. We need support for an additional package manager manifest feature in
+order for this feature to be used by Swift packages.
+
+## Proposed solution
+
+We will add support to the package manifest declaration to specify a set of
+supported Swift language versions:
+
+```swift
+let package = Package(
+    name: "HTTP",
+    ...
+    swiftLanguageVersions: [3, 4])
+```
+
+When building a package, we will always select the Swift language version that
+is most close to (but not exceeding) the major version of the Swift compiler in
+use.
+
+If a package does not support any version compatible with the current compiler,
+we will report an error.
+
+If a package does not specify any Swift language versions, the
+language version to be used will match the major version of the the
+package's Swift tools version (as discussed in a separate evolution proposal). A
+Swift tools version with a major version of '3' will imply a default Swift
+language version of '3', and a Swift tools version with a major version
+of '4' will imply a default Swift language version of '4'.
+
+## Detailed design
+
+We are operating under the assumption that for the immediate future, the Swift
+language compatibility version accepted by the compiler will remain an integer
+major version.
+
+With this change, the complete package initializer will be:
+
+```swift
+    public init(
+        name: String,
+        pkgConfig: String? = nil,
+        providers: [SystemPackageProvider]? = nil,
+        targets: [Target] = [],
+        products: [Product] = [],
+        dependencies: [Dependency] = [],
+        swiftLanguageVersions: [Int]? = nil,
```

Comment:
It might be more future proof to have the api be something like 

```swift
...
swiftLanguageVersions: [SwiftVersion]? = nil,
``` 

with an Int literal convertible for the major version.